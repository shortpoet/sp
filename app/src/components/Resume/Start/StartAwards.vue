<template>
  <section class="resume-section p-3 px-lg-5 d-flex flex-column" id="awards">
    <div class="my-auto">
      <h2 class="heading heading-2 mb-4 mb-md-3">Natural Languages</h2>
      <ul class="fa-ul mb-0">
        <li
          v-for="(lang, i) in spokenLanguages"
          :key="i"
        >
          <i :class="lang.type + ' language-icon fa fa-flag fa-li'"></i>
          <span class="mb-5">{{ `${lang.languages} (${recase(lang.type)})` }} </span>
        </li>
        <li>
          <i class="award-icon fa fa-trophy fa-li"></i>
          Certified Interpreter
        </li>
      </ul>
    </div>
  </section>
</template>

<script>
import _recase from '@/utils/recase.js'
export default {
  name: 'StartAwards',
  props: {
    spokenLanguages: {
      type: Array
    }
  },
  methods: {
    recase(input) {
      return _recase(input)
    }
  }
}
</script>

<style scoped>

</style>