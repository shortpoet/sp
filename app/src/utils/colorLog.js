const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
