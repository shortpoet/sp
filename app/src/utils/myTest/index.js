import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
