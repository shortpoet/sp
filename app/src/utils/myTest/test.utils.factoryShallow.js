import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow