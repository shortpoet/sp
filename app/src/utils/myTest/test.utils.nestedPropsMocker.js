

// // this is incomplete
// // maybe try and develop using TDD  
  
// export const nestedPropsMocker = (props) => {

//   console.log(props)

//   let propsData = {}


//   const handleArray = (props) => {
//     propsData = {}
//     props.map(prop => {
//       propsData[`${prop}`] = `sample ${prop}`
//     })
//     return {
//       propsData
//     }
//   }


//   const handleNested = (props, propsData, previousKey, nextKey) => {
//     console.log(props, propsData)

//     let firstLevel
//     if (previousKey === null && nextKey === null) {
//       firstLevel = true
//     }

//     // spread to clone so the splice works on subsequent tests
//     let values = [...props]

//     const length = parseInt(values.splice(-1, 1)[0])
//     console.log(length)
//     const iterator = [...Array(length).keys()]

//     if (firstLevel === true) {
//       previousKey = values.splice(0, 1)[0]
//       propsData[`${previousKey}`] = []
//     } else {
//       console.log('#### not first run ####')
//       console.log(propsData)
//       console.log(previousKey)
//       nextKey = values.splice(0, 1)[0]
//       console.log(nextKey)
//       propsData[`${previousKey}`][`${nextKey}`] = []
//     }


//     let keyDict

//     for (let i of iterator) {
//       keyDict = {}
//       values.map(prop => {
//         if (Array.isArray(prop)) {
//           console.log(prop)
//           console.log('#### calling handleNested from nested ####')
//           // this would only work for one level nested
//           // TODO maybe refactor for infinite recursion
//           console.log(previousKey)
//           console.log(nextKey)
//           console.log(propsData)
//           handleNested(prop, propsData, previousKey, nextKey)
//         } else {
//           console.log(prop)
//           keyDict[`${prop}`] = `sample ${prop}`
//         }
//       })
//       if (firstLevel === true) {
//         propsData[`${previousKey}`].push(keyDict)
//         firstRun = false
//       } else {
//         console.log(propsData)
//         console.log(previousKey)
//         console.log(nextKey)
//         console.log(propsData.experiences)
//         console.log(propsData.experiences.jobs)
//         propsData[`${previousKey}`][`${nextKey}`].push(keyDict)
//       }
//     }
//     return {
//       propsData
//     }
//   }

//   const isNested = props.some(p => Array.isArray(p))
//   const previousKey = null
//   const nextKey = null

//   if (isNested) {
//     console.log('#### is nested ####')
//     handleNested(props, propsData, previousKey, nextKey)
//   } else  {
//     handleArray(props)
//   }
// }
