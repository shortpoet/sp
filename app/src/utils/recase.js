const recase = (field) => {
  // make sure to leave quotes off regex
  let regex = new RegExp(/([A-Z][a-z])/g)
  let replacement = ' $1'
  let transformation = field.replace(regex, replacement).replace(new RegExp(/^./), s => s.toUpperCase())
  return transformation
}

export default recase
