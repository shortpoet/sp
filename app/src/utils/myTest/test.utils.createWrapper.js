import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper