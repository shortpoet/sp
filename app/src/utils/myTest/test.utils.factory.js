import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory