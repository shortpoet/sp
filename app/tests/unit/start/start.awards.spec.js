import * as myTest from '@/utils/myTest'
const { propsFinder, createWrapper } = myTest.default
import StartAwards from '@/components/Resume/Start/StartAwards'
import recase from '@/utils/recase.js'

describe('StartAwards.vue', () => {

  const component = StartAwards
  const props = Object.keys(component.props)
  let prop
  let selector
  let mockProp = false
  const wrapper = createWrapper(component, propsFinder(props))

  beforeEach(() => {

  })

  it('renders awards h2', () => {

    // hard coded header
    expect(wrapper.find("h2").text()).toMatch("Natural Languages")

  })
  const {propsData} = propsFinder(props)
  const languageTypes = propsData.spokenLanguages.map(l => l.type)

  let iconClass
  let spanText

  
  languageTypes.forEach(type => {
    it(`renders language icons that match spoken languages - ${type} - types/levels prop`, () => {
      // match first word and recase camel to capitalize and add space
      iconClass = recase(wrapper.find(`i.${type}`).attributes().class.match(/^\w+/)[0])
      // match wordSSS .+ vs \w+ within parentheses
      spanText = wrapper.find(`i.${type} ~ span`).text().match(/\((.+)\)/)[1]
      expect(iconClass).toContain(spanText)
    })  
  })

  it('matches snapshot', () => {
    
    expect(wrapper.html()).toMatchSnapshot()

  })

})
