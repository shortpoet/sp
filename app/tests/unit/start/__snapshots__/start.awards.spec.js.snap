// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartAwards.vue matches snapshot 1`] = `
<section id="awards" class="resume-section p-3 px-lg-5 d-flex flex-column">
  <div class="my-auto">
    <h2 class="heading heading-2 mb-4 mb-md-3">Natural Languages</h2>
    <ul class="fa-ul mb-0">
      <li><i class="fluent language-icon fa fa-flag fa-li"></i> <span class="mb-5">Mandarin Chinese, French (Fluent) </span></li>
      <li><i class="workingKnowledge language-icon fa fa-flag fa-li"></i> <span class="mb-5">German, Portuguese (Working Knowledge) </span></li>
      <li><i class="proficient language-icon fa fa-flag fa-li"></i> <span class="mb-5">Italian (Proficient) </span></li>
      <li><i class="native language-icon fa fa-flag fa-li"></i> <span class="mb-5">English, Spanish, Tagalog (Native) </span></li>
      <li><i class="award-icon fa fa-trophy fa-li"></i>
        Certified Interpreter
      </li>
    </ul>
  </div>
</section>
`;
